acquia_lift.install

```diff
@@ -639,6 +638,186 @@ function acquia_lift_update_7017() {
 }
 
 /**
+ * Miscellaneous changes required for the new workflow, includes converting legacy
+ * campaigns to "personalizations".
+ */
+function acquia_lift_update_7018() {
+  // The queue is now gone so delete the queue class variable.
+  variable_del('queue_class_acquia_lift_sync');
+
+  // Get all old-style agents and convert them where possible.
+  $result = db_query("SELECT a.machine_name, a.label, o.plugin, o.osid, o.decision_name, o.options, o.targeting FROM {personalize_agent} a INNER JOIN {personalize_option_sets} o ON o.agent = a.machine_name WHERE a.plugin IN ('acquia_lift', 'acquia_lift_simple_ab')");
+  // First organize our results into arrays of agents with their option sets.
+  $agents = array();
+  foreach ($result as $row) {
+    $options = unserialize($row->options);
+    $agents[$row->machine_name][] = array(
+      'agent_name' => $row->machine_name,
+      'agent_label' => $row->label,
+      'osid' => $row->osid,
+      'decision_name' => empty($row->decision_name) ? 'osid-' + $row->osid : $row->decision_name,
+      'option_count' => count($options),
+      'options' => $options,
+      'targeting' => unserialize($row->targeting),
+    );
+  }
+  // If an agent has more than one option set or if its option set contains fixed
+  // targeting rules, it cannot be converted and will be treated as a legacy agent.
+  $legacy_agents = $convertibles = array();
+  // If personalize module is not enabled all campaigns will be treated as legacy.
+  $personalize_enabled = module_exists('personalize');
```

Comment:
Yep, update functions get run for disabled modules too. Nothing to do with Site Factory, that's just how we first encountered the problem, because there were SF sites that had tried out Lift, created some campaigns, then disabled the modules.